llama-index-integrations/llms/llama-index-llms-sambanova/llama_index/llms/sambanova/base.py

```diff
@@ -0,0 +1,536 @@
+                    chunk["result"] = {"responses": [{"stream_token": ""}]}
+                    return chunk
+                yield chunk
+        except Exception as e:
+            raise RuntimeError(f"Error processing streaming response: {e}")
+
+    def _send_sambaverse_request(
+        self, endpoint: str, data: Dict[str, Any], stream: bool = False
+    ) -> requests.Response:
+        url = self._get_full_url(endpoint)
+        headers = {
+            "key": self.sambaverse_api_key,
+            "Content-Type": "application/json",
+            "modelName": self.sambaverse_model_name,
+        }
+        try:
+            return self._client.post(url, headers=headers, json=data, stream=stream)
+
+        except Exception as e:
+            raise ValueError(f"Error sending request to Sambaverse: {e}")
+
+    def _prepare_request_data(self, prompt: str) -> Dict[str, Any]:
+        try:
+            model_params = self._get_model_kwargs(stop=None)
+            return {"instance": prompt, "params": json.loads(model_params)}
+
+        except Exception as e:
+            raise ValueError(f"Error preparing request data: {e}")
+
+    def _get_completion_from_response(self, response: Dict) -> str:
+        try:
+            return (
+                response.get("result", {})
+                .get("responses", [{}])[0]
+                .get("completion", "")
+            )
+        except Exception as e:
+            raise ValueError(f"Error processing response: {e}")
+
+    @classmethod
+    def class_name(cls) -> str:
+        return "Samabaverse"
+
+    @property
+    def metadata(self) -> LLMMetadata:
+        """LLM metadata."""
+        return LLMMetadata(
+            model_name=self.sambaverse_model_name,
+            model_kwargs=self.model_kwargs,
+            description="Sambanova LLM",
+            is_streaming=self.streaming,
+        )
+
+    @llm_completion_callback()
+    def complete(self, prompt: str) -> CompletionResponse:
+        # Implement the method to complete a prompt
+        data = self._prepare_request_data(prompt)
+        response = self._send_sambaverse_request("api/predict", data)
+        processed_response = self._process_api_response(response)
+        completion_text = self._get_completion_from_response(processed_response)
+
+        return CompletionResponse(text=completion_text)
+
+    @llm_completion_callback()
+    def stream_complete(self, prompt: str) -> CompletionResponseGen:
+        # Implement the method to stream completion
+        # Need to be improved
+        print("In stream_complete")
+        data = self._prepare_request_data(prompt)
+        response = self._send_sambaverse_request("api/predict", data, stream=True)
+
+        for token in self._process_api_stream_response(response):
+            processed_token = token["result"]["responses"][0]["stream_token"]
+            yield CompletionResponse(text=processed_token)
+
+
+class SambaStudio(CustomLLM):
```

Comment:
feel free to put this in its own submodule: `llama_index/llms/sambanova/studio.py`

Reply:
Hi @nerdai  I find it challenging to separate base.py into individual class modules and then inherit from them into a single class, mainly because there are very few common methods or parameters between Sambaverse and SambaStudio. Additionally, there's a risk of users misconfiguring parameters if they are combined in a single class. Therefore, I plan to keep the current implementation as it is. If you have any recommendations or would like to discuss potential updates, please let me know. I also haven’t seen similar implementations in the Llama-index LLMs.